.github/workflows/try_labels.yml

```diff
@@ -0,0 +1,172 @@
+    concurrency:
+      group: try-${{ github.event.number }}
+    outputs:
+      configuration: ${{ steps.configuration.outputs.result }}
+      try_string: ${{ steps.try_string.outputs.result }}
+    steps:
+      - name: Get Try string
+        uses: actions/github-script@v6
+        id: try_string
+        with:
+          script: |
+            function makeComment(body) {
+                console.log(body);
+
+                if (context.eventName != 'pull_request_target')
+                  return;
+
+                github.rest.issues.createComment({
+                  issue_number: context.issue.number,
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  body
+                })
+            }
+
+            let try_string = "${{ inputs.try_string }}" || "";
+
+            if (context.eventName == 'pull_request_target') {
+              for (const label of context.payload.pull_request.labels) {
+                if (!label.name.startsWith("T-")) {
+                  continue;
+                }
+
+                // Try to remove the label. If that fails, it's likely that another
+                // workflow has already processed it or a user has removed it.
+                try {
+                  await github.rest.issues.removeLabel({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    issue_number: context.issue.number,
+                    name: label.name,
+                  });
+                } catch (exception) {
+                  console.log("Assuming '" + label.name + "' is already removed: " + exception);
+                  continue;
+                }
+
+                console.log("Found label: " + label.name);
+                try_string += " " + label.name.slice(2); // "T-" removed
+              }
+            }
+
+            console.log(try_string);
+
+            // empty try string
+            if (!try_string) {
+              return "";
+            }
+
+            let username = context.payload.sender.login;
+            let result = await github.rest.repos.getCollaboratorPermissionLevel({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              username
+            });
+            if (!result.data.user.permissions.push) {
+              makeComment('🔒 User @' + username + ' does not have permission to trigger try jobs.');
+              return "";
+            }
+
+            const url = context.serverUrl +
+              "/" + context.repo.owner +
+              "/" + context.repo.repo +
+              "/actions/runs/" + context.runId;
+            const formattedURL = "[#" + context.runId + "](" + url + ")";
+            makeComment("🔨 Triggering try run (" + formattedURL + ")");
+
+            return try_string;
+      - uses: actions/checkout@v3
+        if: github.event_name != 'issue_comment' && github.event_name != 'pull_request_target'
```

Comment:
I think that the `issue_comment` bit here can be removed as we don't support triggering jobs via issues any longer.